packages/steps/src/BuildStep.ts

```diff
@@ -360,6 +370,27 @@ export class BuildStep extends BuildStepOutputAccessor {
       toJSON: (value: unknown) => JSON.stringify(value),
     };
   }
+
+  private setCommandSpawnTimeouts(spawnPromise: SpawnPromise<SpawnResult>): void {
+    this.commandWarnTimeout = setTimeout(() => {
+      this.ctx.logger.warn(
+        'Command takes longer then expected and it did not produce any logs in the past 15 minutes. Consider evaluating your command for possible issues'
+      );
+    }, BUILD_STEP_WARN_TIMEOUT_MS);
+
+    this.commandKillTimeout = setTimeout(async () => {
+      this.commandTimedOut = true;
+      this.ctx.logger.error(
+        'Command takes a very long time and it did not produce any logs in the past 30 minutes. Most likely an unexpected error happened which caused the process to hang and it will be terminated'
```

Comment:
```suggestion
        'Command takes a very long time and it did not produce any logs in the past 30 minutes. Most likely an unexpected error happened which caused the process to hang and it will be terminated.'
```